person.rb

```diff
@@ -1,26 +1,67 @@
-class Person
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
+
+class Person < Nameable
+  attr_accessor :name, :age
+  attr_reader :id
+
   def initialize(age, name = 'Unknown', parent_permission: true)
+    super()
     @id = rand(1..10_000)
     @name = name
     @age = age
     @parent_permission = parent_permission
   end
 
-  # Getter method for id
-  attr_reader :id
-
-  # Getter and setter method for name, age
-  attr_accessor :name, :age
+  def correct_name
+    @name
+  end
 
-  # define the public method first
+  # define the public method
   def can_use_services?
-    is_of_age? || @parent_permission
+    of_age? || @parent_permission
   end
 
   # define private method is_of_age
   def of_age?
     @age >= 18
   end
 
-  private :is_of_age?
+  private :of_age?
 end
+
+class NameableBaseDecorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < NameableBaseDecorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < NameableBaseDecorator
+  def correct_name
+    @nameable.correct_name
+    @nameable.correct_name[0...10] if @nameable.correct_name.length > 10
+  end
+end
```

Comment:
- [ ] This method returns `nil` for names less than 10 in length. Kindly fix that.

  **SUGGESTION:**
  ```
  def correct_name
    name = @nameable.correct_name.length > 10 ? @nameable.correct_name[0...10] : @nameable.correct_name
    name
  end
  ```
  ![ma](https://user-images.githubusercontent.com/36542986/188833263-e9902843-a349-4bd3-88b1-e41e04a1ee40.jpg)

- [ ] **[OPTIONAL]** I would recommend you move this class to its own file in keeping with Ruby conventions.